iocore/net/quic/QUICStream.h

```diff
@@ -176,7 +177,7 @@ class QUICStream : public VConnection, public QUICFrameGenerator, public QUICTra
  * - no flow control
  * - no state (never closed)
  */
-class QUICCryptoStream : public QUICFrameGenerator
+class QUICCryptoStream : public QUICFrameGenerator, public QUICFrameRetransmitter
```

Comment:
I can live with it, but it's so sad. QUICStream and QUICCryptoStream should definitely have a parent class like `QUICBaseStream`.

Reply:
Yep. Actually it is my next plan to split it into QUICRecvStream、QUICBidStream etc.

```
 23 class QUICStream : public QUICAckReceivedReactor, public QUICDataHolder
 24 {
 25 public:
 26   QUICStream(QUICStreamManager *stream_manager, QUICApplication *app, QUICStreamId sid = 0, bool init_app = true);
 27   virtual ~QUICStream() {}
 28



 90 // QUICSendStream
 91 class QUICSendStream : public QUICStream
 92 {
 93 public:


 64 // QUICCryptoStream
 65 class QUICCryptoStream : public QUICStream
 66 {
 67 public:


106 // QUICRecvStream
107 class QUICRecvStream : public QUICStream
108 {
109 public:


122 // QUICBidirectionalStream
123 class QUICBidirectionalStream : public QUICStream
124 {
125 public:

```

Reply:
When client use send only stream the server should open corresponding recv stream .

Reply:
> Yep. Actually it is my next step to split it into QUICRecvStream、QUICBidStream etc.

Nice. I haven't think about it so much, so I'm not sure at all, but there would be two options. 1) Make them separated classes, and 2) just have a direction property. I think 1) would be better because QUICNetVConnection has a direction property but the code is getting ugly.